src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeRecipientService.kt

```diff
@@ -19,4 +20,8 @@ class BarcodeRecipientService(
     ).let {
       barcodeRecipientRepository.save(it)
     }
+
+  fun getBarcodeRecipient(barcode: Barcode): BarcodeRecipient =
+    barcodeRecipientRepository.getByBarcode(barcode)
+      ?: throw ResourceNotFoundException("Barcode recipient not found for barcode ${barcode.code}")
```

Comment:
Not sure we should throw here... Do we have some early barcodes without recipients?

Reply:
I'll do some research / db queries to support this.
Basically my thinking was that the code that saves the barcode recipient when creating the barcode has been there for a while (I think it was there on our Stocken prod release - SLM-122, code committed 10/02/22). Therefore every barcode has a barcode recipient.
I wrote it this way to handle a null check, and then what would we do if there was no barcode recipient? To me it felt like a very exceptional circumstance, so throwing an exception felt right?

Having said that I accept there could be the use case where someone is scanning a very old barcode that has no barcode recipient record (though not sure we have any of those in production?)

Actually, the more I think about this, the more possible cases I see where there might not be a barcode recipient ....

I'll give this some thought .... 🤔